build: missing use of `JANSSON_CFLAGS` in Makefile?

When building with a side-install copy of jansson, @wihobbs got the following error:
```
../../../../src/common/libpmi/upmi.c:16:10: fatal error: 'jansson.h' file not found
#include <jansson.h>
         ^~~~~~~~~~~
../../../../src/common/libpmi/upmi_single.c:16:10: fatal error: 'jansson.h' file not found
#include <jansson.h>
         ^~~~~~~~~~~
```

It doesn't appear that `JANSSON_CFLAGS` appears in `src/common/libpmi/Makefile.am`, and perhaps it should if `upmi.c` tries to include this header.

--- a/src/common/libpmi/Makefile.am
+++ b/src/common/libpmi/Makefile.am
@@ -9,7 +9,8 @@ AM_CPPFLAGS = \
 	-I$(top_srcdir) \
 	-I$(top_srcdir)/src/include \
 	-I$(top_srcdir)/src/common/libccan \
-	-I$(top_builddir)/src/common/libflux
+	-I$(top_builddir)/src/common/libflux \
+	$(JANSSON_CFLAGS)
 
 noinst_LTLIBRARIES = \
 	libpmi_client.la \

